blockchain/scriptval.go

```diff
@@ -227,7 +244,19 @@ func checkBlockScripts(block *dcrutil.Block, utxoView *UtxoViewpoint, txTree boo
 	}
 	txValItems := make([]*txValidateItem, 0, numInputs)
 	for _, tx := range txs {
-		for txInIdx, txIn := range tx.MsgTx().TxIn {
+		// Skip revocations if the automatic ticket revocations agenda is active and
+		// the transaction version is greater than or equal to 2.  This is allowed
+		// since consensus rules enforce that revocations MUST pay to the addresses
+		// specified by the original commitments in the ticket.
+		msgTx := tx.MsgTx()
+		if isAutoRevocationsEnabled &&
+			msgTx.Version >= stake.TxVersionAutoRevocations &&
+			stake.IsSSRtx(msgTx, isAutoRevocationsEnabled) {
```

Comment:
Good idea, though I need to check that it is guaranteed that `tree` is always set in this case.  It is probably also not a safe assumption in the similar `ValidateTransactionScripts` case above, in which the caller just passes in a `*dcrutil.Tx`, since there is nothing enforcing that `tree` must be set on the passed tx (though it probably is set in all current cases since `mempool` sets it).

Reply:
Realized you were referring to the `txTree` param that is already passed in to this function which simplifies things.  Added the check to short circuit as recommended, good call!